File: python/algo/sorting/bubble_sort.py
import sys


def bubble_sort(items: list) -> None:
    while True:
        swapped = False
        for n in range(0, len(items) - 1):
            if items[n] > items[n + 1]:
                items[n], items[n + 1] = items[n + 1], items[n]
                swapped = True
        if not swapped:
            break

def main() -> None:
    items = [int(x) for x in sys.stdin.readline().split()]
    bubble_sort(items)
    print(" ".join(map(str, items)))


if __name__ == "__main__":
    main()

File: golang/algo/sorting/bubble_sort.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func parseInput(in string) []int {
	arrayStr := strings.SplitN(in, " ", -1)
	var array = []int{}
	for _, i := range arrayStr {
		j, err := strconv.Atoi(i)
		if err != nil {
			panic(err)
		}
		array = append(array, j)
	}
	return array
}

func toString(array []int) string {
	var arrayStr = []string{}
	for _, i := range array {
		j := strconv.Itoa(i)
		arrayStr = append(arrayStr, j)
	}
	return strings.Join(arrayStr, " ")
}

func bubbleSort(array []int) {
	for {
		swapped := false
		for i := 0; i < len(array)-1; i++ {
			if array[i] > array[i+1] {
				tmp := array[i]
				array[i] = array[i+1]
				array[i+1] = tmp
				swapped = true
			}
		}
		if !swapped {
			break
		}
	}
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	in := scanner.Text()
	array := parseInput(in)
	bubbleSort(array)
	fmt.Println(toString(array))
}
